test/integration/device-api/actions.spec.ts

```diff
@@ -608,3 +610,26 @@ describe('manages application lifecycle', () => {
 		});
 	});
 });
+
+describe('reboots or shuts down device', () => {
+	// Stub external dependencies
+	before(async () => {
+		stub(dbus, 'reboot').resolves();
```

Comment:
Could you query the fake dbus API instead of stubbing the method here?

Reply:
This is what I wanted to do originally, but the main unintended side effect is that [there is a listener event that stops the API on 'shutdown'](https://github.com/balena-os/balena-supervisor/blob/fc15ad2554a21b9e86321fc183aab09ba149e15a/src/supervisor.ts#L75). This stops the API of the `balena_supervisor` container in the integration test environment, so it would become necessary to add logic to connect to the Docker socket that manages `balena_supervisor` instead of the `dind` socket, and the spec would need to issue a restart of the `balena_supervisor` service. 

Perhaps instead of the 'shutdown' listener stopping the API, we could use something like https://www.npmjs.com/package/exit-hook to process async exit hooks? This will avoid stopping the API in the test env.

Reply:
Hmm, and who sends the `shutdown` event to the device state, is it a system event? If that's the case the system event should not be triggered by the fake dbus API.

Reply:
It's a system event, and I was wrong about my previous comment. I must have been mistaken from when I was implementing the reboot/shutdown action tests in a different way.

I changed the stubs for reboot/shutdown to spies, since there's no API for querying the test dbus. See 85392f2a858ef9956f211c98c38db1e69b4d1d2d

Reply:
I was actually thinking you could query the status of the service via the DBUS API. I'm not going to keep pushing on this point as I don't want to delay this PR any longer and this looks great anyway